client/state/notices/actions.js

```diff
@@ -20,14 +20,24 @@ export function removeNotice( noticeId ) {
 }
 
 function createNotice( status, text, options = {} ) {
+	const noticeId = options.id || uniqueId();
+	const closeFn = removeNotice.bind( noticeId );
 	const notice = {
-		noticeId: options.id || uniqueId(),
-		duration: options.duration,
+		noticeId,
+		icon: options.icon || null,
+		duration: parseInt( options.duration ) || null,
```

Comment:
I don't know how babel handles it, but for safety reasons its always good to include radix while using parseint: http://stackoverflow.com/questions/6611824/why-do-we-need-to-use-radix